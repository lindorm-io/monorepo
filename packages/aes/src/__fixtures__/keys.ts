import { Kryptos, KryptosFromB64 } from "@lindorm/kryptos";

const defaults = {
  notBefore: new Date("2023-01-01T01:00:00.000Z"),
  updatedAt: new Date("2024-01-01T01:00:00.000Z"),
  expiresAt: new Date("2024-03-01T00:00:00.000Z"),
  issuer: "https://test.lindorm.io/",
  jwksUri: "https://test.lindorm.io/.well-known/jwks.json",
};

const EC: KryptosFromB64 = {
  ...defaults,
  id: "9e568fdb-ed87-5a24-800a-44e1a143d5da",
  curve: "P-521",
  createdAt: new Date("2023-01-01T00:00:00.000Z"),
  privateKey:
    "MIHuAgEAMBAGByqGSM49AgEGBSuBBAAjBIHWMIHTAgEBBEIATAYo3DQYroDV5EgJSUs_kOIvnEScZen73gXa5oQkub0ekQmgOJdPQjINUsdYRn67QK_oBdNUhVbtG_qdxqIgarehgYkDgYYABACN2PIVpRTdfXLmNxkg8Bk2m5netqYsNW2Lefhklr2jfJiVUJUDPoZoGfabGzHEgsKjP2HRbPsEI_tND3x4N9VW0QAID2UYXa7GN0izHWIFRdjVYuR5-0jywFtd-o_N2POdrvlV8xumdVK-TiSPEIdfKoL_Iu0e7IKTsJsj-UmE8rDJnw",
  publicKey:
    "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQAjdjyFaUU3X1y5jcZIPAZNpuZ3ramLDVti3n4ZJa9o3yYlVCVAz6GaBn2mxsxxILCoz9h0Wz7BCP7TQ98eDfVVtEACA9lGF2uxjdIsx1iBUXY1WLkeftI8sBbXfqPzdjzna75VfMbpnVSvk4kjxCHXyqC_yLtHuyCk7CbI_lJhPKwyZ8",
  type: "EC",
};

export const TEST_EC_KEY = Kryptos.from("b64", {
  ...EC,
  algorithm: "ECDH-ES",
  operations: ["encrypt", "decrypt"],
  use: "enc",
});

const OCT: KryptosFromB64 = {
  ...defaults,
  id: "c124f42f-5a61-582d-9fdd-b834de3336a2",
  createdAt: new Date("2023-01-01T00:10:00.000Z"),
  privateKey: "NjVhNF9EK0BTUWJAd0AwUU9IQWkmdTgwWEo2QDJmYjI",
  type: "oct",
};

export const TEST_OCT_KEY = Kryptos.from("b64", {
  ...OCT,
  algorithm: "dir",
  operations: ["encrypt", "decrypt", "deriveKey"],
  use: "enc",
});

const RSA: KryptosFromB64 = {
  ...defaults,
  id: "8bb9bd86-2e70-5cd0-a1cc-b60a5eb6d574",
  createdAt: new Date("2023-01-01T00:30:00.000Z"),
  privateKey:
    "MIIJKQIBAAKCAgEApq21wR1t9ZnC0R2vgmUrDGjq0G5h5THrtC4kdKfavu2t7iwV0RHsQCJU_uboan63dQysFlYLrqvUluMKBgb8lNAGRc1M7LvGKh3zQlV1gtkh71oVlRqUpuGvY6FbBSCVZThC3oRXSlKpRXSMuHY_ohmsXSkZCDReTZevLmzTG7r_bDp8mK05moNU2dfBhh7KtTssyp6etufDrpxsBXF2i-2VzhwMX2GZJY_7DouXgHWTsLIOLqH18gNMUU8RYULNqiA8vBsubVWV9ygr6wLxdZslm4yAXfMVPj6dQ7FCwo4L7gxtqK3M5Olv1C7rRi0NpTCFuUZlOnufWcrSToPKRDvAFUkZPyk9qzyiww5ckVW3ZRaPpVF36RJN-VEU2dskfw2Md2NrMV3W8LNhXryU004CSjJy6bqmer2cQ9UwbZR8y9n4Vxf45lntZMaZ3ZOYnaXuHZL-j3AcZ82bRsApcQneRhJHElA-nJicuZ4s22-B_rvOBu8YNOR5PutAGZz7J5hGr8L-wGDHktqG0ChY30RRlbvK94XitwzXljhBeDmf5SRRoUVbba8PcF91c7IHlb5q4wWxOgEXM9kb0_Oik6iXCGfZVog8DB8uYI6P2Ta5pvMY_Jt_5U-ULu3P8MPNsfHtY0n1Kpf6MP6srdz2kZupT9VL_mReSv3korv7XHMCAwEAAQKCAgBSyOuBt6LCPfUHlSVzb9kiSEf8bx67TgCiVNqxgB9C_l0JcycA2NSNBCBQHJLqNfbbsYgVbfZOcrTbnnd_h9_S78F10OK4oAjsKUUFuNltPHhA3-7hSDr16hkJNH63PLhG9lssFvIeDZFaWGl-QukXE4ZLHRr8Orpt8n1FYzK1TrsHX8M7t6npJvqxWhqaCVgoDHSSTJdLBDDqXHHAgFrfZAkoPkEc7KY6hFkvH8EHQh6xP7aQjcvQwLeamNlqg9oPegSAb0WvczdRc0fg6WzRO1ZB_FZTk2UFLedBVFzysceWG-EpSu198r2i1wzvJXhmCAQbAi5MDlOWrwimi4mdbejnH_YbYJPvAnld5kpUgAvF3p-plAHaMHDA1hp8GrPcPh9QH5gmy5XEbdN7aV6ekAhMRIczxfWnzOc21l3nemK2bpGlSequBzQsAteYk5jYVBYOnfcTwzhS5nS1LPGJ_AVJEdyvJeM3C1WfC-nNuIdBPEvaQzeD8Re9-IY9Az_x_jW9HzeOzXfIkuNfqSyQeOmgUBIQRvyRg1Xfh0KQGp4Ihf2IcqajOo8-D8ZEUvHaheyEHdRXrORH0BFLIQ8UqmC4XjVOx7q3FURgXuqtFoGDx8cGvfHTLKozWrRhtS9DKCeEk-oGKBrv8cYF0-oebbSm2538Dji2XeobJWyldQKCAQEA13wEi30WTdjeMXVWUfgJkFfcApRMnXQHmtNDzS30qao96YNC5yPzrZyfFFhUoOWCi5fmOAMdMtS7BL6ZJp9aroMAi8IhccYdt8rEHxdbMR5pbQ_TqsH1wYW64ETFViR9rPsg5b8CvjmPxW-1FMwl5uOwmrC7sxUrCSTk1PglZVj1ALR2Z1rUjBKQLQFsh1iFBZepxGGD3kbeVqt04d_lYPSewyLGACvSx1BLbR5y8KBNVSialbb9n2sFKSL6gJytr-dHAMdPIM9SQGawrym0ZiEK5FxYsQUaVP5ulw0FrjzwafXBkFjl2FUjwFE3bnRhaR8BGbJAD8xqZarYSzhEHQKCAQEAxgSAqhB71wfDx57gxT8dnTzKNNsklBrMA3Cl3Zip2EG4L1pS1Qd_mL9LjIfF-zgWTMbGW_SQhoi1Fq4MSUFw5R30HP1qbcHw__WnzqLHYVRpJvIQGfe3LRx-EijblfAmKRWUsbvkn-bpwHrrJSm41I5jjmvam366t6NWLBiar4YAMZ-HTSv5-rJkJMsqGCRZHdregCMGSauIXddDZv-gr9OZnpQRX8CTe72tNdkOjGrknB6gxWJxcV3wU08uwUVwfiXDbqmT8SfFEtpAI_SKxXqDWCjJpKT-cxC5QesHapNEHQWO7M0MfsytejU6QmNTESRlb6P8Y5vlcTWMG8kdzwKCAQEAr2Tizici4HqFYPyBJnceUb8x9WeAdSjch8kuhvPBEdjgTBHrgvhggA_b7FN5jv1BJR6e5Tijq5w8HcCGKEnNJf67kwaQ8mqwBcmlmSkJsRFv_rXDNFpsDvkrAyD2_wb0xLgkLt0q_msb-zjRy_vfXfA277KrFTlFrku5dyAEUFn4EnrRZY8pMlppwDhIfmIu5D5ukMpO674De8hTewcjTafPEkZbpAgHqPRScJ84L2RLjeVe7j3kcr5JlRJDlCvztBosURXp8mrOKF1tsyXfEMIUe4wvJOdiZ4wxuAGBfjFb_uNqinPRoiSq1P9Gf0F7iHUUEDZmrnVUpSxKoe9KgQKCAQA4kJa0feryXPx3LJJmpgXuoavdd3A2goUewnIPIAn8haF4nWOmCPNRDewl468ReU8ddXa6U91pIOoh5WwLDQ-T03-lPFrHqawPYfwiketEiVlUT7d7w4Bm-Yml5Ff9t9Ee0VRdhR_k8ZB6dGJxhj6PMXGNAJA89PGffaAjDpA5bZzmj_yVded9zrG-tO58YP24Aoqmrdqg9sgIjnJpnMIpF9f9Lazc3mUC5wbwp-V5BRvy4OjqGZnwWXZKRgdEHHLrBhySowxoygSIwyGpY7S0fq_AYj5fsuYgtnlFND81GYZOiATw7g4QtqUjkMropfjXn77-Kijxx1m-gh5Y3ZWPAoIBAQCOmEBewD34QEnOioz0fZCzWnzd718k-ac7jbjbmTDM6dD6ZGS0RiG-6VfGqPy2D5wJvd3_pV68urq9pCcA6YErcXsXnRuGu1dX7Qu0-D-eHL7XsXYovzd3lbhfp31KP2wnbhwMnp81FO1wrHY9FvRHmIvWd7OE4z8JWDlspLJVqIFgcqOZJCHBYbz32nY8qkr2KhneJPJ4XibBje5xvRTiF_qTPAa633peDgoVCHkg8Ub1b6X9vg_VXSDeJy3EXlRYEGlfuhEfkSSg9oAYuI-XqnCGqj9JffrncWMq7TY5jduveOAyAZwk0FFsHrius3JUoV-aXlueJYKEFG0MmMcP",
  publicKey:
    "MIICCgKCAgEApq21wR1t9ZnC0R2vgmUrDGjq0G5h5THrtC4kdKfavu2t7iwV0RHsQCJU_uboan63dQysFlYLrqvUluMKBgb8lNAGRc1M7LvGKh3zQlV1gtkh71oVlRqUpuGvY6FbBSCVZThC3oRXSlKpRXSMuHY_ohmsXSkZCDReTZevLmzTG7r_bDp8mK05moNU2dfBhh7KtTssyp6etufDrpxsBXF2i-2VzhwMX2GZJY_7DouXgHWTsLIOLqH18gNMUU8RYULNqiA8vBsubVWV9ygr6wLxdZslm4yAXfMVPj6dQ7FCwo4L7gxtqK3M5Olv1C7rRi0NpTCFuUZlOnufWcrSToPKRDvAFUkZPyk9qzyiww5ckVW3ZRaPpVF36RJN-VEU2dskfw2Md2NrMV3W8LNhXryU004CSjJy6bqmer2cQ9UwbZR8y9n4Vxf45lntZMaZ3ZOYnaXuHZL-j3AcZ82bRsApcQneRhJHElA-nJicuZ4s22-B_rvOBu8YNOR5PutAGZz7J5hGr8L-wGDHktqG0ChY30RRlbvK94XitwzXljhBeDmf5SRRoUVbba8PcF91c7IHlb5q4wWxOgEXM9kb0_Oik6iXCGfZVog8DB8uYI6P2Ta5pvMY_Jt_5U-ULu3P8MPNsfHtY0n1Kpf6MP6srdz2kZupT9VL_mReSv3korv7XHMCAwEAAQ",
  type: "RSA",
};

export const TEST_RSA_KEY = Kryptos.from("b64", {
  ...RSA,
  algorithm: "RSA-OAEP-256",
  operations: ["encrypt", "decrypt"],
  use: "enc",
});
