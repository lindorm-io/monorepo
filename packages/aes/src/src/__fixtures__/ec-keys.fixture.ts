import { Kryptos } from "@lindorm/kryptos";

export const EC_KEY_SET = Kryptos.from("pem", {
  id: "b1ca2923-7a82-5f21-8d23-8d76cf4e2abf",
  curve: "P-521",
  privateKey:
    "-----BEGIN PRIVATE KEY-----\n" +
    "MIHuAgEAMBAGByqGSM49AgEGBSuBBAAjBIHWMIHTAgEBBEIB+1dzKp5FO/wISTaI\n" +
    "PHCv1yZLMZl0o6BAXgMKibC6e8iFI9W4YIrZYTInHDpqA/wNd764XikWWse9jRRw\n" +
    "KLYy02KhgYkDgYYABAFyw1SdEKD9Sz2XojfjqCPVoilq5YSZjLBmZtXG2x6ydRUW\n" +
    "3Q/IL7kfEtKycJ42z9UhP4adevymNBQ4J02/Yx/jswGcKMep28r825kUE64LbsUc\n" +
    "8b4/pgKHEX2bR9n1Vw5uJRWEtPKVBXeV9dk0D1I/1/iDYJRkc3OwZ1+MYWN31i5r\n" +
    "tg==\n" +
    "-----END PRIVATE KEY-----\n",
  publicKey:
    "-----BEGIN PUBLIC KEY-----\n" +
    "MIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQBcsNUnRCg/Us9l6I346gj1aIpauWE\n" +
    "mYywZmbVxtsesnUVFt0PyC+5HxLSsnCeNs/VIT+GnXr8pjQUOCdNv2Mf47MBnCjH\n" +
    "qdvK/NuZFBOuC27FHPG+P6YChxF9m0fZ9VcObiUVhLTylQV3lfXZNA9SP9f4g2CU\n" +
    "ZHNzsGdfjGFjd9Yua7Y=\n" +
    "-----END PUBLIC KEY-----\n",
  type: "EC",
});
