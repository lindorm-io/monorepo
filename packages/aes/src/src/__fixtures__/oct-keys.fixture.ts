import { Kryptos } from "@lindorm/kryptos";

export const OCT_KEY_SET = Kryptos.from("pem", {
  id: "e3ff08aa-af8a-5e3b-9608-6aad72f8fe06",
  privateKey: "aIDH@Vq@FOvVHEngSd@Erjlw@SnsAtMu",
  type: "oct",
});
