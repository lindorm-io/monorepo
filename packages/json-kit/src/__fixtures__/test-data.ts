export const TEST_DICT = {
  one: [
    true,
    new Date("2020-01-01T08:00:00.000Z"),
    12345,
    "string",
    {
      thirteen: [true, 123, "arr", null, undefined],
      fourteen: false,
      fifteen: new Date("2020-01-01T08:00:00.000Z"),
      sixteen: 123,
      seventeen: "str",
    },
  ],
  two: true,
  three: new Date("2020-01-01T08:00:00.000Z"),
  four: 12345,
  five: {
    six: "a",
    seven: 1,
    eight: null,
    nine: true,
    ten: [],
    eleven: undefined,
  },
  twelve: "string",
  thirteen: Buffer.from("hello"),
};

export const TEST_DICT_STRING = `{"__meta__":{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S","thirteen":"F"},"__record__":{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string","thirteen":"aGVsbG8"}}`;

export const TEST_ARRAY = [TEST_DICT, TEST_DICT];

export const TEST_ARRAY_STRING = `{"__meta__":[{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S","thirteen":"F"},{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S","thirteen":"F"}],"__array__":[{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string","thirteen":"aGVsbG8"},{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string","thirteen":"aGVsbG8"}]}`;
