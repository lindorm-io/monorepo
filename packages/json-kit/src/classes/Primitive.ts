import { isArray, isObject, isString } from "@lindorm/is";
import { Dict } from "@lindorm/types";
import { _getMetaArray } from "../utils/private/get-meta-array";
import { _getMetaObject } from "../utils/private/get-meta-object";
import { _parseArrayValues } from "../utils/private/parse-array-values";
import { _parseObjectValues } from "../utils/private/parse-object-values";
import { _stringifyArrayValues } from "../utils/private/stringify-array-values";
import { _stringifyObjectValues } from "../utils/private/stringify-object-values";

export class Primitive<T extends Array<any> | Dict = Dict> {
  private readonly _data: Array<any> | Dict;
  private readonly _meta: Array<any> | Dict;

  public constructor(input: any) {
    if (isArray(input)) {
      this._data = _stringifyArrayValues(input);
      this._meta = _getMetaArray(input);
    } else if (isObject(input)) {
      this._data = _stringifyObjectValues(input);
      this._meta = _getMetaObject(input);
    } else if (isString(input)) {
      const { data, meta } = JSON.parse(input);

      this._data = data;
      this._meta = meta;
    } else {
      throw new TypeError("Expected input to be an array or object");
    }
  }

  // getters

  public get data(): Array<any> | Dict {
    return this._data;
  }

  public get meta(): Array<any> | Dict {
    return this._meta;
  }

  // public

  public toJSON(): T {
    return Primitive.parse(this._data, this._meta) as T;
  }

  public toString(): string {
    return JSON.stringify({ json: this._data, meta: this._meta });
  }

  // private

  private static parse(input: any, meta: any): Array<any> | Dict {
    return isArray(input)
      ? _parseArrayValues(input, meta)
      : _parseObjectValues(input, meta);
  }
}
