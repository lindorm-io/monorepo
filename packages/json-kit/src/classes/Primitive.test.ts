import MockDate from "mockdate";
import { TEST_ARRAY, TEST_DICT, TEST_STRING } from "../__fixtures__/test-data";
import { Primitive } from "./Primitive";

MockDate.set("2020-01-01T08:00:00.000Z");

describe("Primitive", () => {
  describe("array", () => {
    let struct: Primitive;

    beforeAll(() => {
      struct = new Primitive(TEST_ARRAY);
    });

    test("should resolve expected data", () => {
      expect(struct.data).toMatchSnapshot();
    });

    test("should resolve expected meta", () => {
      expect(struct.meta).toMatchSnapshot();
    });

    test("should resolve expected json", () => {
      expect(struct.toJSON()).toMatchSnapshot();
    });

    test("should resolve expected string", () => {
      expect(struct.toString()).toMatchSnapshot();
    });
  });

  describe("dict", () => {
    let struct: Primitive;

    beforeAll(() => {
      struct = new Primitive(TEST_DICT);
    });

    test("should resolve expected data", () => {
      expect(struct.data).toMatchSnapshot();
    });

    test("should resolve expected meta", () => {
      expect(struct.meta).toMatchSnapshot();
    });

    test("should resolve expected json", () => {
      expect(struct.toJSON()).toMatchSnapshot();
    });

    test("should resolve expected string", () => {
      expect(struct.toString()).toMatchSnapshot();
    });
  });

  describe("string", () => {
    let struct: Primitive;

    beforeAll(() => {
      struct = new Primitive(TEST_STRING);
    });

    test("should resolve expected data", () => {
      expect(struct.data).toMatchSnapshot();
    });

    test("should resolve expected meta", () => {
      expect(struct.meta).toMatchSnapshot();
    });

    test("should resolve expected json", () => {
      expect(struct.toJSON()).toMatchSnapshot();
    });

    test("should resolve expected string", () => {
      expect(struct.toString()).toMatchSnapshot();
    });
  });
});
