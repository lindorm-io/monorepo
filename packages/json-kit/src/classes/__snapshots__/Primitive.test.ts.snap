// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitive array should resolve expected data 1`] = `
[
  {
    "five": {
      "eight": 0,
      "eleven": 0,
      "nine": "true",
      "seven": "1",
      "six": "a",
      "ten": [],
    },
    "four": "12345",
    "one": [
      "true",
      "2020-01-01T08:00:00.000Z",
      "12345",
      "string",
      {
        "fifteen": "2020-01-01T08:00:00.000Z",
        "fourteen": "false",
        "seventeen": "str",
        "sixteen": "123",
        "thirteen": [
          "true",
          "123",
          "arr",
          0,
          0,
        ],
      },
    ],
    "three": "2020-01-01T08:00:00.000Z",
    "twelve": "string",
    "two": "true",
  },
  {
    "five": {
      "eight": 0,
      "eleven": 0,
      "nine": "true",
      "seven": "1",
      "six": "a",
      "ten": [],
    },
    "four": "12345",
    "one": [
      "true",
      "2020-01-01T08:00:00.000Z",
      "12345",
      "string",
      {
        "fifteen": "2020-01-01T08:00:00.000Z",
        "fourteen": "false",
        "seventeen": "str",
        "sixteen": "123",
        "thirteen": [
          "true",
          "123",
          "arr",
          0,
          0,
        ],
      },
    ],
    "three": "2020-01-01T08:00:00.000Z",
    "twelve": "string",
    "two": "true",
  },
]
`;

exports[`Primitive array should resolve expected json 1`] = `
[
  {
    "five": {
      "eight": null,
      "eleven": undefined,
      "nine": true,
      "seven": 1,
      "six": "a",
      "ten": [],
    },
    "four": 12345,
    "one": [
      true,
      2020-01-01T08:00:00.000Z,
      12345,
      "string",
      {
        "fifteen": 2020-01-01T08:00:00.000Z,
        "fourteen": false,
        "seventeen": "str",
        "sixteen": 123,
        "thirteen": [
          true,
          123,
          "arr",
          null,
          undefined,
        ],
      },
    ],
    "three": 2020-01-01T08:00:00.000Z,
    "twelve": "string",
    "two": true,
  },
  {
    "five": {
      "eight": null,
      "eleven": undefined,
      "nine": true,
      "seven": 1,
      "six": "a",
      "ten": [],
    },
    "four": 12345,
    "one": [
      true,
      2020-01-01T08:00:00.000Z,
      12345,
      "string",
      {
        "fifteen": 2020-01-01T08:00:00.000Z,
        "fourteen": false,
        "seventeen": "str",
        "sixteen": 123,
        "thirteen": [
          true,
          123,
          "arr",
          null,
          undefined,
        ],
      },
    ],
    "three": 2020-01-01T08:00:00.000Z,
    "twelve": "string",
    "two": true,
  },
]
`;

exports[`Primitive array should resolve expected meta 1`] = `
[
  {
    "five": {
      "eight": "L",
      "eleven": "U",
      "nine": "B",
      "seven": "N",
      "six": "S",
      "ten": [],
    },
    "four": "N",
    "one": [
      "B",
      "D",
      "N",
      "S",
      {
        "fifteen": "D",
        "fourteen": "B",
        "seventeen": "S",
        "sixteen": "N",
        "thirteen": [
          "B",
          "N",
          "S",
          "L",
          "U",
        ],
      },
    ],
    "three": "D",
    "twelve": "S",
    "two": "B",
  },
  {
    "five": {
      "eight": "L",
      "eleven": "U",
      "nine": "B",
      "seven": "N",
      "six": "S",
      "ten": [],
    },
    "four": "N",
    "one": [
      "B",
      "D",
      "N",
      "S",
      {
        "fifteen": "D",
        "fourteen": "B",
        "seventeen": "S",
        "sixteen": "N",
        "thirteen": [
          "B",
          "N",
          "S",
          "L",
          "U",
        ],
      },
    ],
    "three": "D",
    "twelve": "S",
    "two": "B",
  },
]
`;

exports[`Primitive array should resolve expected string 1`] = `"{"json":[{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string"},{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string"}],"meta":[{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S"},{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S"}]}"`;

exports[`Primitive dict should resolve expected data 1`] = `
{
  "five": {
    "eight": 0,
    "eleven": 0,
    "nine": "true",
    "seven": "1",
    "six": "a",
    "ten": [],
  },
  "four": "12345",
  "one": [
    "true",
    "2020-01-01T08:00:00.000Z",
    "12345",
    "string",
    {
      "fifteen": "2020-01-01T08:00:00.000Z",
      "fourteen": "false",
      "seventeen": "str",
      "sixteen": "123",
      "thirteen": [
        "true",
        "123",
        "arr",
        0,
        0,
      ],
    },
  ],
  "three": "2020-01-01T08:00:00.000Z",
  "twelve": "string",
  "two": "true",
}
`;

exports[`Primitive dict should resolve expected json 1`] = `
{
  "five": {
    "eight": null,
    "eleven": undefined,
    "nine": true,
    "seven": 1,
    "six": "a",
    "ten": [],
  },
  "four": 12345,
  "one": [
    true,
    2020-01-01T08:00:00.000Z,
    12345,
    "string",
    {
      "fifteen": 2020-01-01T08:00:00.000Z,
      "fourteen": false,
      "seventeen": "str",
      "sixteen": 123,
      "thirteen": [
        true,
        123,
        "arr",
        null,
        undefined,
      ],
    },
  ],
  "three": 2020-01-01T08:00:00.000Z,
  "twelve": "string",
  "two": true,
}
`;

exports[`Primitive dict should resolve expected meta 1`] = `
{
  "five": {
    "eight": "L",
    "eleven": "U",
    "nine": "B",
    "seven": "N",
    "six": "S",
    "ten": [],
  },
  "four": "N",
  "one": [
    "B",
    "D",
    "N",
    "S",
    {
      "fifteen": "D",
      "fourteen": "B",
      "seventeen": "S",
      "sixteen": "N",
      "thirteen": [
        "B",
        "N",
        "S",
        "L",
        "U",
      ],
    },
  ],
  "three": "D",
  "twelve": "S",
  "two": "B",
}
`;

exports[`Primitive dict should resolve expected string 1`] = `"{"json":{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string"},"meta":{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S"}}"`;

exports[`Primitive string should resolve expected data 1`] = `
{
  "five": {
    "eight": 0,
    "eleven": 0,
    "nine": "true",
    "seven": "1",
    "six": "a",
    "ten": [],
  },
  "four": "12345",
  "one": [
    "true",
    "2020-01-01T08:00:00.000Z",
    "12345",
    "string",
    {
      "fifteen": "2020-01-01T08:00:00.000Z",
      "fourteen": "false",
      "seventeen": "str",
      "sixteen": "123",
      "thirteen": [
        "true",
        "123",
        "arr",
        0,
        0,
      ],
    },
  ],
  "three": "2020-01-01T08:00:00.000Z",
  "twelve": "string",
  "two": "true",
}
`;

exports[`Primitive string should resolve expected json 1`] = `
{
  "five": {
    "eight": null,
    "eleven": undefined,
    "nine": true,
    "seven": 1,
    "six": "a",
    "ten": [],
  },
  "four": 12345,
  "one": [
    true,
    2020-01-01T08:00:00.000Z,
    12345,
    "string",
    {
      "fifteen": 2020-01-01T08:00:00.000Z,
      "fourteen": false,
      "seventeen": "str",
      "sixteen": 123,
      "thirteen": [
        true,
        123,
        "arr",
        null,
        undefined,
      ],
    },
  ],
  "three": 2020-01-01T08:00:00.000Z,
  "twelve": "string",
  "two": true,
}
`;

exports[`Primitive string should resolve expected meta 1`] = `
{
  "five": {
    "eight": "L",
    "eleven": "U",
    "nine": "B",
    "seven": "N",
    "six": "S",
    "ten": [],
  },
  "four": "N",
  "one": [
    "B",
    "D",
    "N",
    "S",
    {
      "fifteen": "D",
      "fourteen": "B",
      "seventeen": "S",
      "sixteen": "N",
      "thirteen": [
        "B",
        "N",
        "S",
        "L",
        "U",
      ],
    },
  ],
  "three": "D",
  "twelve": "S",
  "two": "B",
}
`;

exports[`Primitive string should resolve expected string 1`] = `"{"json":{"one":["true","2020-01-01T08:00:00.000Z","12345","string",{"thirteen":["true","123","arr",0,0],"fourteen":"false","fifteen":"2020-01-01T08:00:00.000Z","sixteen":"123","seventeen":"str"}],"two":"true","three":"2020-01-01T08:00:00.000Z","four":"12345","five":{"six":"a","seven":"1","eight":0,"nine":"true","ten":[],"eleven":0},"twelve":"string"},"meta":{"one":["B","D","N","S",{"thirteen":["B","N","S","L","U"],"fourteen":"B","fifteen":"D","sixteen":"N","seventeen":"S"}],"two":"B","three":"D","four":"N","five":{"six":"S","seven":"N","eight":"L","nine":"B","ten":[],"eleven":"U"},"twelve":"S"}}"`;
